packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx

```diff
@@ -0,0 +1,515 @@
+          <div class="item-controls">
+            {showItemsPerPageControl ? (
+              <div class="items-per-page-holder">
+                <ic-typography
+                  class={{
+                    [`pagination-text-${appearance}`]: true,
+                    ["items-per-page-control-label"]: true,
+                  }}
+                  variant="label"
+                >
+                  {this.itemLabel}s per {this.pageLabel.toLowerCase()}
+                </ic-typography>
+                <ic-select
+                  small
+                  label="items-per-page-input"
+                  class="items-per-page-input"
+                  hideLabel
+                  options={itemsPerPageOptions}
+                  value={this.itemsPerPage}
+                  onIcChange={() => this.changeItemsPerPage()}
+                ></ic-select>
+              </div>
+            ) : null}
+            {showItemsPerPage && paginationType === "data" ? (
+              <ic-typography
+                class={{
+                  [`pagination-text-${appearance}`]: true,
+                  ["item-pagination-label"]: true,
+                }}
+                variant="label"
+                aria-live="polite"
+              >
+                {this.lowerBound} - {this.upperBound} of {this.totalItems}{" "}
+                {this.itemLabel.toLowerCase()}
+                {this.totalItems > 1 ? "s" : ""}
+              </ic-typography>
+            ) : showItemsPerPage ? (
+              <ic-typography
+                class={{
+                  [`pagination-text-${appearance}`]: true,
+                  ["page-pagination-label"]: true,
+                }}
+                variant="label"
+                aria-live="polite"
+              >
+                {this.pageLabel} {this.currentPage} of {this.totalPages}
+              </ic-typography>
+            ) : null}
+          </div>
+        ) : null}
+        <div
+          class={{
+            ["pagination-controls"]: true,
+            ["pagination-controls-wrap"]: this.paginationWrapped,
+          }}
+        >
+          <div class="pagination-holder">
+            <ic-pagination
+              appearance={appearance}
+              type={paginationControl}
+              pages={this.totalPages}
+            ></ic-pagination>
+          </div>
+          {showGoToPageControl ? (
+            <div class="go-to-page-holder">
+              <ic-typography
+                class={{ [`pagination-text-${appearance}`]: true }}
+                variant="label"
+              >
+                Go to {this.pageLabel.toLowerCase()}
+              </ic-typography>
+              <ic-tooltip
+                label={this.inputError}
+                target="#go-to-page-input"
+                disableHover
+                disableClick
+              >
+                <ic-text-field
+                  type="number"
+                  small
+                  label="go-to-page-input"
```

Comment:
'go-to-page-input' string is used in a few places to can be private variable at the top of the page.